Stripe/STPAPIClient.m

```diff
@@ -403,6 +406,47 @@ - (void)createSourceWithParams:(STPSourceParams *)sourceParams completion:(STPSo
                                        }];
 }
 
+- (void)precheckSourceWithParams:(STPSourcePrecheckParams *)precheckParams
+                      completion:(STPSourcePrecheckCompletionBlock)completion {
+    NSCAssert(precheckParams != nil, @"'precheckParams' is required for precheck");
+    NSCAssert(precheckParams.sourceID != nil, @"'precheckParams.sourceID' is required for precheck");
+    NSCAssert(completion != nil, @"'completion' is required to use the precheck result that is created");
+
+    NSString *precheckEndpoint = [[sourcesEndpoint stringByAppendingPathComponent:precheckParams.sourceID] stringByAppendingPathComponent:precheckFinalPathComponent];
+    NSDictionary *params = [STPFormEncoder dictionaryForObject:precheckParams];
+    [STPAPIRequest<STPSourcePrecheckResult *> postWithAPIClient:self
+                                                       endpoint:precheckEndpoint
+                                                     parameters:params
+                                                     serializer:[STPSourcePrecheckResult new]
+                                                     completion:^(STPSourcePrecheckResult *object, __unused NSHTTPURLResponse *response, NSError *error) {
+                                                         completion(object, error);
+                                                     }];
+}
+
+- (void)precheckAndCreate3DSSourceIfNeededWithSourceParams:(STPSourceParams *)sourceParams
+                                                completion:(STP3DSSourcePrecheckCompletionBlock)completion {
+    NSCAssert(sourceParams != nil, @"'params' is required to create a source");
+    NSCAssert(completion != nil, @"'completion' is required to use the source that is created");
+    NSCAssert(sourceParams.type == STPSourceTypeThreeDSecure, @"Must use 3DS params");
+    NSCAssert(sourceParams.threeDSecureParamsCardId != nil, @"Must have a card id");
+
+    STPSourcePrecheckParams *precheckParams = [STPSourcePrecheckParams new];
```

Comment:
Yeah, you can set additional parameters if you do the two requests separately, but the convenience method doesnt do it. I'm not sure the best way to resolve this, maybe actually just delete this convenience method.